### ISSUE TITLE
doc patch - process.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4269]
|<img alt="teamnoir's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| teamnoir reported|
|-|-|
|Trac ID|trac#4269|
|Type|defect|
|Created|2010-02-10 19:46:41Z|

I'm not certain, but it looks like the example is broken.  It looks as though the close is happening multiple times during the "for" loop when it should be happening once after the for loop.

Attachments:

* [pxs](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/05c/05c97210a34828d9365b1e656dd2e4cb1fa885dc/b53bb29da8299ee7a307681b561147222c5cbb2c) (736 bytes) - added by teamnoir on 2010-02-10 19:47:01Z - doc patch

<details><summary>Searchable metadata</summary>

```
trac-id__4269 4269
type__defect defect
reporter__teamnoir teamnoir
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1265831201000000 1265831201000000
changetime__1265988854000000 1265988854000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="teamnoir's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|teamnoir removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

For various reasons, it turns out that this doesn't actually matter.  `loseConnection` will wait until the send buffer has been emptied before actually tearing down the connection.  Combined with the fact that `transport.write` actually just adds to the send buffer and no attempt is made to empty the send buffer until the reactor regains control means that all the bytes we want to write to stdin will actually get written to it before the connection is lost.

However, it's probably not what the author _intended_ to write, and it's less confusing the way you've patched it.  So I'm going to apply your patch.  Thanks. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [28176]) Apply process docs code listing patch fixing indentation


Author: teamnoir
Reviewer: exarkun
Fixes: [#4269](https://github.com/twisted/twisted/issues/4269)

De-dent the `closeStdin` call in the process example so that it is outside the loop
which is writing to stdin.  This doesn't actually change any behavior, but it's
more clear this way, and removes the reliance on the send buffer not being flushed
until the reactor regains control.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
